framework/tlvf/tlvf.py

```diff
@@ -1006,7 +1006,16 @@ def addClassVarLenMethods(self, obj_meta, param_type, param_name, param_meta, pa
             lines_cpp.append( "%sreturn false;" % self.getIndentation(2))
             lines_cpp.append( "%s}" % self.getIndentation(1) )
 
-            lines_cpp.append( "%sif (ptr->getStartBuffPtr() != (uint8_t *)m_%s) {" % (self.getIndentation(1), param_name) )
+            lines_cpp.append("%suint8_t *src = (uint8_t*) m_%s;" % (self.getIndentation(1), param_meta.name))
+            lines_cpp.append("%sif (!m_parse__) {" %self.getIndentation(1))
+            if param_length != None:
+                lines_cpp.append("%sif (m_%s_idx__ != 0) {" % (self.getIndentation(2), param_name))
+                ptr = "(uint8_t *)m_%s_vector[m_%s_idx__ - 1]->getBuffPtr()" %(param_meta.name,param_name)
+                lines_cpp.append("%ssrc = %s;" %(self.getIndentation(3), ptr))                    
```

Comment:
If this wasn't generated code, I wouldn't even have a `m_%s_idx__` member and instead use:
```suggestion
                lines_cpp.append("%sif (m_%s_vector) {" % (self.getIndentation(2), param_name))
                lines_cpp.append("%ssrc = (uint8_t *)m_%s_vector.back()->getBuffPtr();" % self.getIndentation(3)
```
But maybe there are reasons to prefer to use `m_%s_idx__`

Reply:
I agree, `m_%s_idx__` is mainly leftovers of old code (back in the days when we only supported simple lists), and now it's only needed for simple unknown length lists.
However, I think that changing the code to use `m_%s_idx__` only when necessary exceeds the scope of this PR and should be separated.